File: LATIHAN_ARRAY/rs/src/c_array/c1.rs
#[derive(Debug)]
pub struct ArrayInfo<T> {
    pub length: usize,
    pub first: Option<T>,
    pub last: Option<T>,
    pub is_empty: bool,
}

pub fn get_array_info<T: Clone>(arr: &[T]) -> ArrayInfo<T> {
    ArrayInfo {
        length: arr.len(),
        first: arr.first().cloned(),
        last: arr.last().cloned(),
        is_empty: arr.is_empty(),
    }
}

File: challenge/LATIHAN_ARRAY/js/c1.js
// Tulis function getArrayInfo di sini
function getArrayInfo(arr) {
    return {
        length: arr.length,
        first: arr[0],
        last: arr[arr - 1],
        isEmpty: arr.length === 0
    }
}
console.log(getArrayInfo([1, 2, 3, 4, 5]));
console.log(getArrayInfo(["a", "b", "c"]));
console.log(getArrayInfo([]));